src/github.ts

```diff
@@ -2221,6 +2221,106 @@ export class GitHub {
       });
     }
   }
+
+  /**
+   * Waits for a GitHub release to be listed by repeatedly polling the GitHub API.
+   */
+  async waitForReleaseToBeListed({tagName, id}: GitHubRelease) {
+    for (let attempt = 0; attempt < 10; attempt++) {
+      this.logger.debug(
+        `Checking if release ${tagName} is listed on GitHub (attempt ${
+          attempt + 1
+        })...`
+      );
+
+      const releases = await this.octokit.repos.listReleases({
+        owner: this.repository.owner,
+        repo: this.repository.repo,
+        page: 1,
+        per_page: 10,
+      });
+      let found = false;
+      for (const release of releases.data) {
+        if (release.id === id) {
+          found = true;
+          break;
+        }
+      }
+      if (found) {
+        this.logger.debug(`Release ${tagName} listed on GitHub`);
+        return;
+      }
+
+      await sleepInMs(500 * attempt);
+    }
+
+    this.logger.warn(`Release ${tagName} is not yet listed on GitHub`);
+  }
+
+  /**
+   * Waits for a file in a given branch to meet a specified condition, checking up to 10 times.
+   *
+   * @param {Function} params.checkFileStatus - A function that takes an object with either `fileContent` and `kind: 'success'`
+   *                                            or `error` and `kind: 'error'`. It should return a boolean indicating
+   *                                            whether the file is considered up to date and we should stop waiting.
+   *
+   * @throws Will throw if an error other than FileNotFound occurs while fetching the file.
+   **/
+  async waitForFileToBeUpToDateOnBranch({
+    branch,
+    filePath,
+    checkFileStatus,
+  }: {
+    branch: string;
+    filePath: string;
+    checkFileStatus: (
+      arg:
+        | {fileContent: string; kind: 'success'}
+        | {error: FileNotFoundError; kind: 'error'}
```

Comment:
This seems to overcomplicate things a bit. I would say, just retry on any error, and on the last retry throw the error
```
for 0..10
  try
    checkFileStatus()
      return
  catch
    log and ignore
if !checkFileStatus()
  warn

Reply:
There isn't really a case where the checkFileStatus would have an intelligent decision to make on file not found

Reply:
Hmm, in fact we already have the octokit retry plugin. The only error we should handle here is the file-not-found, others are already handled by the plugin.